functions/ocr/app/save.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018, Google, LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_ocr_save]
+
+package ocr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+)
+
+// SaveResult is executed when a message is published to the Cloud Pub/Sub topic specified by
+// RESULT_TOPIC in config.json file, and saves the data packet to a file in GCS.
+func SaveResult(ctx context.Context, event PubSubMessage) error {
+	var message ocrMessage
+	if event.Data != nil {
+		messageData := event.Data
+		err := json.Unmarshal(messageData, &message)
+		if err != nil {
+			return fmt.Errorf("json.Unmarshal: %v", err)
+		}
+	} else {
+		return fmt.Errorf("Empty data")
+	}
+	text := message.Text
+	fileName := message.FileName
+	lang := message.Lang
```

Comment:
Probably don't need these variables - can reference directly. Even if the later lines are longer, you don't have to keep these new variables in your head as you're reading.

Reply:
Done